QuizApp/QuizApp/Repositories/QuizRepository.swift

```diff
@@ -15,22 +15,42 @@ protocol QuizRepositoryProtocol {
 class QuizRepository: QuizRepositoryProtocol {
 
     private let quizNetworkDataSource: QuizNetworkDataSourceProtocol
+    private let quizDatabaseDataSource: QuizDatabaseDataSourceProtocol
 
-    init(quizNetworkDataSource: QuizNetworkDataSourceProtocol) {
+    init(quizNetworkDataSource: QuizNetworkDataSourceProtocol, quizDatabaseDataSource: QuizDatabaseDataSourceProtocol) {
         self.quizNetworkDataSource = quizNetworkDataSource
+        self.quizDatabaseDataSource = quizDatabaseDataSource
     }
 
     func fetchQuizzes(for category: CategoryRepoModel) async throws -> [QuizRepoModel] {
-        let quizzes = try await quizNetworkDataSource.fetchQuizzes(
-            for: CategoryDataModel(rawValue: category.rawValue)!)
-        return quizzes
-            .map { QuizRepoModel($0) }
+        do {
+            let quizzes = try await quizNetworkDataSource
+                .fetchQuizzes(for: CategoryDataModel(rawValue: category.rawValue)!)
+                .map { QuizRepoModel($0) }
+            quizDatabaseDataSource.save(quizzes: quizzes.map { QuizDatabaseModel($0) })
+            return quizzes
+        } catch RequestError.disconnectedError {
+            let quizzes = quizDatabaseDataSource
+                .fetchQuizzes()
```

Comment:
Why can't DatabaseDS have `fetchQuizzes(for category)` and use filtering on DB level?